client/src/main/java/ch/bifrost/client/impl/keyexchange/DHKeyExchangeClient.java

```diff
@@ -0,0 +1,47 @@
+package ch.bifrost.client.impl.keyexchange;
+
+import java.math.BigInteger;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Optional;
+
+import ch.bifrost.core.api.datagram.CounterpartAddress;
+import ch.bifrost.core.api.datagram.DatagramEndpoint;
+import ch.bifrost.core.api.datagram.DatagramMessage;
+import ch.bifrost.core.api.keyexchange.DHKeyExchange;
+import ch.bifrost.core.api.keyexchange.IdKeyPair;
+import ch.bifrost.core.api.keyexchange.Key;
+import ch.bifrost.core.api.keyexchange.DHKeyExchangeRequestMessage;
+import ch.bifrost.core.api.keyexchange.DHKeyExchangeResponseMessage;
+
+public class DHKeyExchangeClient implements KeyExchangeClient {
+
+	private DatagramEndpoint datagramEndpoint;
+	private CounterpartAddress serverAddress;
+
+	public DHKeyExchangeClient (DatagramEndpoint datagramEndpoint, CounterpartAddress serverAddress) {
+		this.datagramEndpoint = datagramEndpoint;
+		this.serverAddress = serverAddress;
+	}
+
+	@Override
```

Comment:
Since this implements an interface method and does not override an object method; is this annotation here correct/needed?